lib/ecto/adapters/sqlite3.ex

```diff
@@ -227,6 +227,17 @@ defmodule Ecto.Adapters.SQLite3 do
     end
   end
 
+  @impl Ecto.Adapter.Schema
+  def autogenerate(:id), do: nil
+  def autogenerate(:embed_id), do: Ecto.UUID.generate()
```

Comment:
What is the `:embed_id`? Is that the id on an `embedded_schema`?

Reply:
Yes, it appears to [automatically replace `:binary_id` for embedded schemas](https://github.com/elixir-ecto/ecto/blob/bc59a232c829b9d51acbd4277423f4549e5a6ffc/lib/ecto/embedded.ex#L244). I'm not too sure why this is the case though, I've searched the docs, forum, mailing list, ecto and ecto_sql issues and have barely found any mention of it.

This is also how it's implemented in the [base SQL adapter](https://github.com/elixir-ecto/ecto_sql/blob/520494878635949924a6ae585f7a4d0b84d6a221/lib/ecto/adapters/sql.ex#L174) and the [MSSQL adapter](https://github.com/elixir-ecto/ecto_sql/blob/520494878635949924a6ae585f7a4d0b84d6a221/lib/ecto/adapters/tds.ex#L144).

Thinking more about it, my guess would be that these get generated as strings because they're stored as data, not actual UUID types, so that ensures they remain readable. Should that be the case, I guess it would make sense for us to have them conform to `:binary_id_type`.

What do you think?

Reply:
Yea this looks fine to me. Was just curious about this method and more so how I missed it when implementing the adapter.

Reply:
Yeah the Postgres and MySQL adapters don't implement it, they rely on the base SQL adapter's implementation so I guess that's why you missed it!